x-pack/plugins/elastic_assistant/server/lib/langchain/llm/actions_client_llm.ts

```diff
@@ -77,7 +77,8 @@ export class ActionsClientLlm extends LLM {
     const requestBody = {
       actionId: this.#connectorId,
       params: {
-        subAction: this.#request.body.subAction,
+        // hard code to non-streaming subaction as this class only supports non-streaming
+        subAction: 'invokeAI',
```

Comment:
Is there anything in place to ensure all new gen-ai connectors must implement this subAction? I see separate enum's for [each](https://github.com/elastic/kibana/blob/09222ee908afed926dffb86326491310274be363/x-pack/plugins/stack_connectors/common/openai/constants.ts#L19) [connector](https://github.com/elastic/kibana/blob/ad8163db828c25333797955ec0b870696e90fe54/x-pack/plugins/stack_connectors/common/bedrock/constants.ts#L19), but no common/shared enum. Probably no big deal since each new connector type will be tested with the assistant, but I was just thinking through new connector development.

Reply:
yeah I've had it on my to do list to create a GenAIConnector subclass of SubActionConnector, just like [CaseConnector](https://github.com/elastic/kibana/blob/main/x-pack/plugins/actions/server/sub_action_framework/README.md#classes) that would enforce the method structure, but time has been a constraint